src/kontrol/solc_to_k.py

```diff
@@ -64,6 +64,84 @@ def solc_to_k(
     return _kprint.pretty_print(bin_runtime_definition, unalias=False) + '\n'
 
 
+@dataclass(frozen=True)
+class Input:
+    parent_name: str
+    name: str
+    type: str
+    components: list[Input] = field(default_factory=list)
+
+    @staticmethod
+    def from_dict(_input: dict) -> Input:
+        name = _input['name']
+        type = _input['type']
+        if _input.get('components') is not None:
+            if type == 'tuple':
+                components = Input._recurse_comp(name, _input['components'])
+            else:
+                dimension = _find_array_dimensions(type)
+                components = Input._recurse_comp(name + "[]" * dimension, _input['components'])
```

Comment:
The `flake8` that checks if the source code respects the style guides is setup to throw if double quotes are used. You can run locally in the root of the repo `make format; make` to format the source code and run unit tests.

Reply:
That's why the CI is failing here https://github.com/runtimeverification/kontrol/actions/runs/6611613582/job/17955868147?pr=121 .